tests/Aspire.Hosting.Tests/Utils/LoggerNotificationExtensions.cs

```diff
@@ -77,21 +74,23 @@ await foreach (var resourceEvent in resourceNotificationService.WatchAsync(cance
                 if (loggingResourceIds.Add(resourceId))
                 {
                     // Start watching the logs for this resource ID
-                    logWatchTasks.Add(WatchResourceLogs(tcs, resourceId, logTexts, resourceLoggerService, cancellationToken));
+                    logWatchTasks.Add(WatchResourceLogs(tcs, resourceId, logTexts, resourceLoggerService, cancellationTokenSource));
```

Comment:
Are we using `logWatchTasks` here to root them to the method so they don't get collected?

Reply:
I believe these would just be on the ThreadPool so they wouldn't be collected, but not sure. Feels safer this way, unless someone says it's fine without.